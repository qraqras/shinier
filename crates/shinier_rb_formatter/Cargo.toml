[package]
name = "shinier_rb_formatter"
version = "0.1.0"
edition = "2024"

[dependencies]
ruby-prism = { version = "1.5.1" }
shinier_fs = { workspace = true }

[dev-dependencies]
insta = { version = "1.43.2", features = ["glob"] }
