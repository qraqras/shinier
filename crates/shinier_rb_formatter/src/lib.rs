macro_rules! println_dbg {
    ($fmt:expr) => {
        if cfg!(debug_assertions) {
            (print!(concat!($fmt, "\n")));
        }
    };
}

mod builder;
mod doc;
mod formatter;
mod printer;
mod renderer;

pub use builder::*;
pub use doc::*;
pub use formatter::*;
pub use printer::*;
pub use renderer::*;
