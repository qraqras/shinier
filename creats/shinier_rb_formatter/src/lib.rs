mod doc;
mod formatter;
mod printer;
mod visitor;

pub use doc::*;
pub use formatter::*;
pub use printer::*;
pub use visitor::*;
