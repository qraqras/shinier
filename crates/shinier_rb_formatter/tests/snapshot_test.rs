use insta::{assert_snapshot, glob};
use shinier_rb_formatter::Printer;

#[test]
fn run() {
    glob!("**/*.rb", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let output = Printer::new(contents, ()).print();
        assert_snapshot!(output);
    });
}
