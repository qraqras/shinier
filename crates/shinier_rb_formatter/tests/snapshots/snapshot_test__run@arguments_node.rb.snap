---
source: crates/shinier_rb_formatter/tests/snapshot_test.rs
assertion_line: 9
expression: output
input_file: crates/shinier_rb_formatter/tests/ruby/node/arguments_node.rb
---
func(foo, bar, baz)
func(
  foo,
  bar,
  baz,
  foo,
  bar,
  baz,
  foo,
  bar,
  baz,
  foo,
  bar,
  baz,
  foo,
  bar,
  baz
)
